### PR TITLE
JAMES-2683 Plug the thread leak related to Connection and Channel man…

### PR DESCRIPTION
**Pull request by mbaechler**
For a RabbitMQMailQueueTest run.
Before : 
140 not closed threads, 297 created
After : 
37 not closed threads, 166 created

### CODE REVIEW

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/SimpleConnectionPool.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    private Mono<Connection> getOpenConnection() {
+        Connection previous = connectionReference.get();
+        Connection current = Optional.ofNullable(previous)
+            .filter(Connection::isOpen)
+            .orElseGet(connectionFactory::create);
+        boolean updated = connectionReference.compareAndSet(previous, current);
+        if (updated) {
+            return Mono.just(current);
+        } else {
+            return Mono.error(new RuntimeException("unable to create and register a new Connection"));
```
Comment (rouazana) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/SimpleConnectionPool.java]: you don't need to close current in this case?

Comment (mbaechler) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/SimpleConnectionPool.java]: you are right

**Comment by mbaechler** [comment]
test this please

**Comment by rouazana** [comment]
```
Progress (2): 3.4/9.3 MB | [INFO] There is 1 error reported by Checkstyle 8.5 with checkstyle.xml ruleset.
[33fb25ee864f8e3beb4a08a78f4c2fbab98684b3] [ERROR] src/main/java/org/apache/james/backend/rabbitmq/SimpleConnectionPool.java:[25,8] (imports) UnusedImports: Unused import - java.util.concurrent.TimeoutException.
```

**Comment by rouazana** [comment]
```
[62659ce3b3ee646224e98f9c73b29f8bea49f608] [ERROR] Errors: 
[62659ce3b3ee646224e98f9c73b29f8bea49f608] [ERROR]   VacationIntegrationTest.jmapVacationShouldNotSendNotificationTwice:240 » ConditionTimeout
```
test this please

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/SimpleConnectionPool.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        }
+    }
+
+    public boolean tryConnection() {
+        try {
+            return getOpenConnection()
+                .blockOptional(Duration.ofSeconds(1))
+                .isPresent();
+        } catch (Throwable t) {
+            return false;
```
Comment (aduprat) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/SimpleConnectionPool.java]: You may log here.

**Comment by aduprat** [comment]
It could be great to have metrics on `SimpleConnectionPool` for opened & closed connection (in another issue).

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/SimpleConnectionPool.java (hunk 2 above)
Comment (mbaechler) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/rabbitmq/SimpleConnectionPool.java]: I don't think so. The method *try* if the connection is ok. It's not an error or even a warning if it throws. It's used for healthcheck.

**Comment by mbaechler** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**